### PR TITLE
Adds benchmarking for the main TC scheduler loop.

### PR DESCRIPTION
**Pull request by barath**
Makes a few other minor C -> C++ changes as needed.

### CODE REVIEW

File: core/tc.cc [hunk 1]
```diff
@@ -606,62 +607,66 @@ static inline struct task_result tc_scheduled(struct tc *c) {
...
-      if (unlikely(ctx.is_pause_requested())) {
-        if (unlikely(ctx.Block())) break;
-        last_stats = s->stats;
-        last_print_tsc = checkpoint = now = rdtsc();
-      } else if (unlikely(FLAGS_s && now - last_print_tsc >= tsc_hz)) {
-        print_stats(s, &last_stats);
-        last_stats = s->stats;
-        last_print_tsc = checkpoint = now = rdtsc();
+inline bool schedule_once(uint64_t round, struct sched *s) {
+  static struct sched_stats last_stats = s->stats;
```
Comment (sangjinhan) [inline core/tc.cc]: You should not use local-scoped static variables here, since all worker threads will contend on the same memory address.

Comment (barath) [inline core/tc.cc]: Good point.  Will fix.

**Comment by barath** [comment]
Updated pull request.

**Comment by sangjinhan** [comment]
Merged, thanks! FYI, on an E5-2650 v2 machine:

```
Benchmark                                     Time           CPU Iterations
---------------------------------------------------------------------------
TCFixture/TCScheduleOnceCount/4/0            29 ns         29 ns   20583769
TCFixture/TCScheduleOnceCount/8/0            26 ns         26 ns   27421070
TCFixture/TCScheduleOnceCount/16/0           27 ns         27 ns   25730965
TCFixture/TCScheduleOnceCount/32/0           30 ns         30 ns   23649221
TCFixture/TCScheduleOnceCount/64/0           33 ns         33 ns   21494254
TCFixture/TCScheduleOnceCount/128/0          38 ns         38 ns   18230163
TCFixture/TCScheduleOnceCount/256/0          42 ns         42 ns   15996654
TCFixture/TCScheduleOnceCount/512/0          48 ns         48 ns   13881780
TCFixture/TCScheduleOnceCount/1024/0         55 ns         55 ns   12633890
TCFixture/TCScheduleOnceCount/2k/0           60 ns         60 ns   11641887
TCFixture/TCScheduleOnceCount/4k/0           65 ns         65 ns   10812202
TCFixture/TCScheduleOnceCount/8k/0           69 ns         69 ns   10202571
TCFixture/TCScheduleOnceCount/16k/0          73 ns         73 ns    9476663
TCFixture/TCScheduleOnceCount/32k/0          76 ns         76 ns    9211431
TCFixture/TCScheduleOnceCount/64k/0         120 ns        120 ns    5396659
TCFixture_BigO                             5.75 lgN       5.75 lgN
TCFixture_RMS                                19 %         19 %
TCFixture/TCScheduleOnceCycle/4/1            29 ns         29 ns   24428058
TCFixture/TCScheduleOnceCycle/8/1            31 ns         31 ns   22116566
TCFixture/TCScheduleOnceCycle/16/1           32 ns         32 ns   22157678
TCFixture/TCScheduleOnceCycle/32/1           33 ns         33 ns   21580786
TCFixture/TCScheduleOnceCycle/64/1           35 ns         35 ns   20312332
TCFixture/TCScheduleOnceCycle/128/1          39 ns         39 ns   18222178
TCFixture/TCScheduleOnceCycle/256/1          43 ns         43 ns   16460215
TCFixture/TCScheduleOnceCycle/512/1          49 ns         49 ns   14580025
TCFixture/TCScheduleOnceCycle/1024/1         56 ns         56 ns   12552172
TCFixture/TCScheduleOnceCycle/2k/1           65 ns         65 ns   10989773
TCFixture/TCScheduleOnceCycle/4k/1           74 ns         74 ns    9402567
TCFixture/TCScheduleOnceCycle/8k/1           90 ns         90 ns    7763780
TCFixture/TCScheduleOnceCycle/16k/1         109 ns        109 ns    6465431
TCFixture/TCScheduleOnceCycle/32k/1         126 ns        126 ns    5572042
TCFixture/TCScheduleOnceCycle/64k/1         164 ns        164 ns    4390009
TCFixture_BigO                             7.41 lgN       7.41 lgN
TCFixture_RMS                                26 %         26 %
```

**Pull request merged**